src/videojs-hls.js

```diff
@@ -1148,14 +1160,15 @@ videojs.Hls.getMediaIndexByTime = function() {
  * @returns {number} The current time to that point, or 0 if none appropriate.
  */
 videojs.Hls.prototype.getCurrentTimeByMediaIndex_ = function(playlist, mediaIndex) {
-  var index, time = 0;
+  var index, time = 0, segment;
 
   if (!playlist.segments || mediaIndex === 0) {
     return 0;
   }
 
   for (index = 0; index < mediaIndex; index++) {
-    time += playlist.segments[index].duration;
+    segment = playlist.segments[index];
+    time += segment.preciseDuration || segment.duration || playlist.targetDuration || 0;
```

Comment:
I believe this function got re-added somewhere along the way. It should be removed entirely.